packages/typespec-ts/src/modular/serialization/serializers.ts

```diff
@@ -0,0 +1,245 @@
+      functionType === UsageFlags.Input
+        ? serializerMetadata?.serializerFunctionName
+        : serializerMetadata?.deserializerFunctionName;
+    return {
+      expr: `${
+        functionName ?? functionType === UsageFlags.Input
+          ? "MISSING_SERIALIZER"
+          : "MISSING_DESERIALIZER"
+      }((${valueExpr}))`,
+      imports: functionName
+        ? [{ importType: "serializers", name: functionName }]
+        : undefined
+    };
+  }
+
+  switch (type.type) {
+    case "string":
+    case "constant":
+    case "float":
+    case "integer":
+    case "boolean":
+    case "any":
+      return { expr: valueExpr };
+    case "model": {
+      if (type.name) {
+        // has serialize function, should be unreachable
+        break;
+      }
+      return serializeModelInline(functionType, serializerMap, type, valueExpr);
+    }
+    case "datetime":
+      return { expr: serializeDatetime(functionType, type, valueExpr) };
+    case "list": {
+      return serializeList(functionType, serializerMap, type, valueExpr);
+    }
+    case "byte-array": {
+      return serializeByteArray(functionType, type, valueExpr);
+    }
+    case "combined":
+      if (isSpecialHandledUnion(type)) {
+        // has serialize function, should be unreachable
+        break;
+      }
+      return {
+        expr: isNormalUnion(type) ? valueExpr : `(${valueExpr}) as any`
+      };
+    case "enum":
+      // TODO: encodedName
+      return { expr: valueExpr };
+  }
+  return {
+    expr: "FIXME"
+  };
+}
+
+export function serializeByteArray(
+  functionType: UsageFlags,
+  type: Type,
+  valueExpr: string
+): SerializerOutput {
+  if (type.format === "binary") {
+    return { expr: valueExpr };
+  }
+  const functionName =
+    functionType === UsageFlags.Input
+      ? "uint8ArrayToString"
+      : "stringToUint8Array";
+  const format =
+    functionType === UsageFlags.Input ? getEncodingFormat(type) : type.format;
+  const functionCall = `${functionName}(${valueExpr}, "${format ?? "base64"}")`;
+  const expr =
+    functionType === UsageFlags.Input
+      ? functionCall
+      : `(typeof (${valueExpr}) === 'string')
+            ? (${functionCall})
+            : (${valueExpr})`;
+  const imports: Imports = [{ importType: "coreUtil", name: functionName }];
+  return { expr, imports };
+}
+
+export function serializeDatetime(
+  functionType: UsageFlags,
+  type: Type,
+  valueExpr: string
+): string {
+  if (functionType === UsageFlags.Input) {
+    switch (type.format) {
+      case "date":
+        return `(${valueExpr}).toDateString()`;
+      case "time":
+        return `(${valueExpr}).toTimeString()`;
+      case "rfc7231":
+      case "headerDefault":
+        return `(${valueExpr}).toUTCString()`;
+      case "unixTimestamp":
+        return `(${valueExpr}).getTime()`;
+      case "rfc3339":
+      default:
+        return `(${valueExpr}).toISOString()`;
+    }
+  } else {
+    return `new Date(${valueExpr})`;
```

Comment:
I am not sure why we need create a Date for Datetime serialization ? 
```suggestion
    return `${valueExpr}`;
```